### PR TITLE
[RFC 66] Add custom namespace columns to structural variants

### PR DESCRIPTION
**Pull request by BasLee**
Add custom namespace columns to structural variants data. 

image

# Changes
- add `namespaces` to `meta_structural_variants.txt` and `data_structural_variants.txt`
- add `ANNOTATION_JSON` column to `structural_variant` table 
- add `namespaceColumns` property to structural variants API

# Tests
- TestImportStructuralVariantData.java
- StructuralVariantControllerTest.java

# Checks
- [ ] After importing the es_0 test study, custom namespaces should be returned by the API:
```
curl -s 'http://localhost:8081/api/structural-variant/fetch' \
  -H 'content-type: application/json' \
  --data-raw '{"entrezGeneIds":[57670], "sampleMolecularIdentifiers": [{"molecularProfileId":"study_es_0_structural_variants", "sampleId":"TCGA-A2-A04P-01"}]}' \
  | jq '.[0].namespaceColumns'
```

### CODE REVIEW

File: core/src/test/java/org/mskcc/cbio/portal/scripts/TestImportStructuralVariantData.java [hunk 1]
```diff
@@ -79,4 +82,59 @@ public void testImportStructuralVariantData() throws DaoException, IOException {
         assertEquals(0L, (long) structuralVariants.get(3).getSite2EntrezGeneId()); // null in db
         assertEquals(673L, (long) structuralVariants.get(3).getSite1EntrezGeneId());
     }
+
+    @Test
+    public void testImportStructuralVariantDataImportsCustomNamespacesFromTwoSamples() throws DaoException, IOException {
```
Comment (pvannierop) [inline core/src/test/java/org/mskcc/cbio/portal/scripts/TestImportStructuralVariantData.java:87]: Kan vervangen worden met een parametrized test.

**Review by pvannierop — CHANGES_REQUESTED** [review]
Good work!

File: core/src/test/java/org/mskcc/cbio/portal/scripts/TestImportStructuralVariantData.java (hunk 1 above)
Comment (BasLee) [reply core/src/test/java/org/mskcc/cbio/portal/scripts/TestImportStructuralVariantData.java]: Fixed with @pvannierop in [4dd78fe](https://github.com/cBioPortal/cbioportal/pull/9880/commits/4dd78fe83c92add105ad383052636eacd650e55a)

File: core/src/test/java/org/mskcc/cbio/portal/scripts/TestImportStructuralVariantData.java [hunk 2]
```diff
@@ -79,4 +82,59 @@ public void testImportStructuralVariantData() throws DaoException, IOException {
...
+        Set<String> namespacesToImport = newHashSet("StructVarNamespace", "StructVarNamespace2");
+        ImportStructuralVariantData importer = new ImportStructuralVariantData(file, geneticProfileId, null, namespacesToImport);
+        importer.importData();
+        MySQLbulkLoader.flushAll();
+
+        List<StructuralVariant> all = DaoStructuralVariant
+            .getAllStructuralVariants();
+        
+        // Namespace values present:
+        String annotationJson2 = all
```
[the remark this answers is not in the document]
Comment (BasLee) [reply core/src/test/java/org/mskcc/cbio/portal/scripts/TestImportStructuralVariantData.java]: Fixed in [8c2b8ea](https://github.com/cBioPortal/cbioportal/commit/8c2b8eac27d874b436bddc251cc4655b2be3c038)

File: core/src/main/java/org/mskcc/cbio/portal/util/StructuralVariantUtil.java [hunk 3]
```diff
@@ -32,15 +32,22 @@
...
 import org.mskcc.cbio.portal.model.StructuralVariant;
+import org.mskcc.cbio.portal.scripts.*;
 
 /**
  * @author ochoaa
  */
 public class StructuralVariantUtil {
     private HashMap<String, Integer> columnIndexMap;
+    private final NamespaceColumnParser namespaceColumnParser;
+    private ObjectMapper objectMapper = new ObjectMapper();
```
Comment (Luke-Sikina) [inline core/src/main/java/org/mskcc/cbio/portal/util/StructuralVariantUtil.java:50]: `final`

File: web/src/main/java/org/cbioportal/web/StructuralVariantController.java [hunk 4]
```diff
@@ -66,7 +66,8 @@ public ResponseEntity<List<StructuralVariant>> fetchStructuralVariants(
             @RequestAttribute(required = false, value = "involvedCancerStudies") Collection<String> involvedCancerStudies,
             @Valid @RequestAttribute(required = false, value = "interceptedStructuralVariantFilter") StructuralVariantFilter interceptedStructuralVariantFilter,
             @ApiParam(required = true, value = "List of entrezGeneIds and molecularProfileIds or sampleMolecularIdentifiers")
-            @Valid @RequestBody(required = false) StructuralVariantFilter structuralVariantFilter) {
+            @Valid @RequestBody(required = false) StructuralVariantFilter structuralVariantFilter
```
Comment (Luke-Sikina) [inline web/src/main/java/org/cbioportal/web/StructuralVariantController.java:69]: Let's remove this whitespace change from the git diff

**Review by Luke-Sikina — APPROVED** [review]

File: web/src/main/java/org/cbioportal/web/StructuralVariantController.java (hunk 4 above)
Comment (BasLee) [reply web/src/main/java/org/cbioportal/web/StructuralVariantController.java]: Fixed in [52b680d](https://github.com/cBioPortal/cbioportal/commit/52b680d6bf158a8f0f4308e4fa385921110aa5b8)

File: core/src/main/java/org/mskcc/cbio/portal/util/StructuralVariantUtil.java (hunk 3 above)
Comment (BasLee) [reply core/src/main/java/org/mskcc/cbio/portal/util/StructuralVariantUtil.java]: Fixed in [52b680d](https://github.com/cBioPortal/cbioportal/commit/52b680d6bf158a8f0f4308e4fa385921110aa5b8)

**Review by dippindots — APPROVED** [review]

**Pull request merged**